libs/Microsoft.MixedReality.WebRTC.Unity/Assets/Microsoft.MixedReality.WebRTC.Unity.Examples/StandaloneDemo/HardcodedSignaler.cs

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEngine;
+using UnityEngine.UI;
+using Microsoft.MixedReality.WebRTC.Unity;
+using System.Threading;
+
+/// <summary>
+/// Simple signaler using two peer connections in the same process,
+/// and hard-coding their SDP message delivery to avoid the need for
+/// any kind of networking to deliver SDP messages.
+/// 
+/// This component is designed to be used in demos where both peers
+/// are present in the same scene.
+/// </summary>
+public class HardcodedSignaler : MonoBehaviour
+{
+    /// <summary>
+    /// First peer to connect, which will generate an offer.
+    /// </summary>
+    public PeerConnection Peer1;
+
+    /// <summary>
+    /// Second peer to connect, which will wait for an offer from the first peer.
+    /// </summary>
+    public PeerConnection Peer2;
+
+    /// <summary>
+    /// The button that generates an offer from peer #1 to peer #2.
+    /// </summary>
+    [Tooltip("The button that generates an offer to a given target")]
+    public Button CreateOfferButton;
+
+    private ManualResetEventSlim _remoteApplied1 = new ManualResetEventSlim();
+    private ManualResetEventSlim _remoteApplied2 = new ManualResetEventSlim();
+
+    public void Connect()
+    {
+        _remoteApplied1.Reset();
+        _remoteApplied2.Reset();
+        Peer1.CreateOffer();
+        _remoteApplied1.Wait();
+        _remoteApplied2.Wait();
+    }
+
+    public bool Connect(int millisecondsTimeout)
+    {
+        _remoteApplied1.Reset();
+        _remoteApplied2.Reset();
+        Peer1.CreateOffer();
+        if (!_remoteApplied1.Wait(millisecondsTimeout))
+        {
+            return false;
+        }
+        if (!_remoteApplied2.Wait(millisecondsTimeout))
+        {
+            return false;
+        }
+        return true;
+    }
+
+    private void Start()
+    {
+        Peer1.OnInitialized.AddListener(OnInitialized1);
+        Peer2.OnInitialized.AddListener(OnInitialized2);
+
+        CreateOfferButton?.onClick.AddListener(() =>
+        {
+            Peer1.CreateOffer();
+        });
```

Comment:
nit: this pattern - the component takes a reference to an existing button and adds an onClick listener on start - seems a bit unusual, I'd expect the button to be configured in the inspector to trigger a call to Connect in this component.

Reply:
I tried but it doesn't seem like you can access a non-Unity object like `Peer1` to call a method on it from a Unity event assigned in the editor. This needs to be done from code.

Reply:
Actually, `Peer1` is a component here. Still, the `CreateOffer()` method doesn't show up in the list of thing you can call automatically on button click.

Reply:
I would expect any void method to be available for wiring? Can you simply wire `LocalOnlySignaler.Connect()`?

Reply:
Yes that one works, will fix.

Reply:
Fixed.